lib/mongoid/association/referenced/has_many/enumerable.rb

```diff
@@ -216,6 +216,21 @@ def empty?
               end
             end
 
+            # Does the enumerable have any? Will determine if there exists at least
+            # one element in the enumerable, whether or not it is _loaded.
+            #
+            # @example Doess the enumerable have any?
+            #   enumerable.any?
+            #
+            # @return [ true, false ] If the enumerable has any elements.
+            def any? 
+              if _loaded?
+                in_memory.count > 0
+              else
+                _unloaded.exists? || _added.count > 0
+              end
+            end
```

Comment:
Could you make sure this method works with the `has_and_belongs_to_many` association as well?